src/rustdoc_gen.rs

```diff
@@ -123,11 +182,48 @@ pub(crate) enum CrateType<'a> {
     },
 }
 
+#[derive(Debug, Clone, Hash)]
+pub(crate) struct FeatureConfig {
+    pub(crate) base_features: BaseFeatures,
+    pub(crate) extra_features: Vec<String>,
+    pub(crate) ignore_non_existing: bool,
+}
```

Comment:
Can you document what this type represents and what each of its fields means? It isn't clear to me what the difference between a "base" feature and an "extra" feature is, or when `ignore_non_existing` applies.